File: examples/python/_shared/env.py
"""Environment inputs required to run the CAW agent demos."""

from __future__ import annotations

import os
from dataclasses import dataclass

DEFAULT_DESTINATION = "0x1111111111111111111111111111111111111111"


@dataclass(frozen=True)
class DemoEnv:
    """Credentials and target address every example reads from the environment.

    Load it once at the top of each demo via :meth:`DemoEnv.load` and pass the
    instance around instead of re-reading ``os.environ`` in every helper.
    """

    api_url: str
    api_key: str
    wallet_id: str
    destination: str

    @classmethod
    def load(cls) -> "DemoEnv":
        """Read credentials from the process environment.

        Raises:
            KeyError: if any of ``AGENT_WALLET_API_URL``, ``AGENT_WALLET_API_KEY``
                or ``AGENT_WALLET_WALLET_ID`` is unset.
        """
        return cls(
            api_url=os.environ["AGENT_WALLET_API_URL"],
            api_key=os.environ["AGENT_WALLET_API_KEY"],
            wallet_id=os.environ["AGENT_WALLET_WALLET_ID"],
            destination=os.environ.get("CAW_DESTINATION", DEFAULT_DESTINATION),
        )

File: examples/typescript/lib/env.ts
/**
 * Environment variable loader with fail-fast validation.
 *
 * Every agent example starts by calling `loadEnv()` (via `DemoContext.load`)
 * so that a misconfigured machine produces a single, clear error instead of
 * surfacing as a mysterious 401 / 404 later in the request chain.
 */

export interface DemoEnv {
  /** Base URL of the Cobo Agentic Wallet service, e.g. `https://api.agenticwallet.cobo.com`. */
  basePath: string;
  /** Owner-scoped API key (used to submit pacts and read audit logs). */
  ownerKey: string;
  /** UUID of the wallet the demo operates on. */
  walletId: string;
  /** Address to transfer towards; defaults to a burn-style placeholder. */
  destination: string;
  /** Only required when actually invoking an LLM-backed agent. */
  openaiApiKey?: string;
}

const DEFAULT_DESTINATION = '0x1111111111111111111111111111111111111111';

function requireEnv(name: string): string {
  const value = process.env[name];
  if (!value) {
    throw new Error(
      `Missing required environment variable: ${name}. See README.md for the full list.`,
    );
  }
  return value;
}

export function loadEnv(): DemoEnv {
  return {
    basePath: requireEnv('AGENT_WALLET_API_URL'),
    ownerKey: requireEnv('AGENT_WALLET_API_KEY'),
    walletId: requireEnv('AGENT_WALLET_WALLET_ID'),
    destination: process.env.CAW_DESTINATION ?? DEFAULT_DESTINATION,
    openaiApiKey: process.env.OPENAI_API_KEY,
  };
}
